### PR TITLE
Remove element message dispatcher

### PR DESCRIPTION
**Pull request by andpodob**
Moves handling messages back to the Element module.

### CODE REVIEW

**Comment by mat-hek** [comment]
@andpodob it seems this PR somehow contains changes from https://github.com/membraneframework/membrane_core/pull/288

File: lib/membrane/core/element.ex [hunk 1]
```diff
@@ -119,27 +116,161 @@ defmodule Membrane.Core.Element do
...
+  defp handle_message_error(message, mode, reason, state) do
+    Membrane.Logger.error("""
+    Cannot handle message: #{inspect(message)}, mode: #{inspect(mode)}
+    Reason: #{inspect(reason)}
+    State: #{inspect(state, pretty: true)}
+    """)
+
+    reason = {:cannot_handle_message, reason, message: message, mode: mode}
+
+    {{:error, reason}, state}
```
Comment (mat-hek) [inline lib/membrane/core/element.ex:274]: Please revisit if this is actually useful. `Membrane.Helper.GenServer.(no)reply` already handle logging. I don't think that wrapping error in that `{:cannot_handle_message, ..}` tuple adds any valuable information too.

File: lib/membrane/core/element/action_handler.ex [hunk 2]
```diff
@@ -142,11 +142,11 @@ defmodule Membrane.Core.Element.ActionHandler do
   defp do_handle_action(
          {:demand, {pad_ref, size}},
          cb,
-         params,
+         _params,
          %State{type: type} = state
        )
        when is_pad_ref(pad_ref) and is_demand_size(size) and type in [:sink, :filter] do
-    supply_demand(pad_ref, size, cb, params[:supplying_demand?] || false, state)
```
Comment (mat-hek) [inline lib/membrane/core/element/action_handler.ex:149]: this was already reviewed and merged, wasn't it?

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane/core/element/action_handler.ex (hunk 2 above)
Comment (andpodob) [reply lib/membrane/core/element/action_handler.ex]: Yes, I think so.

File: lib/membrane/core/element.ex [hunk 3]
```diff
@@ -119,27 +115,148 @@ defmodule Membrane.Core.Element do
...
+
+  defp handle_message_error(message, mode, reason, state) do
+    Membrane.Logger.error("""
+    Cannot handle message: #{inspect(message)}, mode: #{inspect(mode)}
+    Reason: #{inspect(reason)}
+    State: #{inspect(state, pretty: true)}
+    """)
+
+    {{:error, reason}, state}
   end
```
Comment (mat-hek) [inline lib/membrane/core/element.ex:261]: is this needed? `noreply` already handles logging

File: lib/membrane/core/element.ex [hunk 4]
```diff
@@ -250,7 +250,7 @@ defmodule Membrane.Core.Element do
     end
   end
 
-  defp handle_message_error(message, mode, reason, state) do
+  defp handle_message_error(_message, _mode, reason, state) do
```
Comment (mat-hek) [inline lib/membrane/core/element.ex:253]: this function is useless now

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane/core/element.ex [hunk 5]
```diff
@@ -186,40 +186,40 @@ defmodule Membrane.Core.Element do
...
   @impl GenServer
-  def handle_info(Message.new(:handle_unlink, pad_ref) = msg, state) do
+  def handle_info(Message.new(:handle_unlink, pad_ref), state) do
     case PadController.handle_unlink(pad_ref, state) do
       {:ok, _state} = res ->
         noreply(res)
 
       {{:error, reason}, new_state} ->
-        handle_message_error(msg, :info, reason, new_state) |> noreply(state)
+        noreply({{:error, reason}, new_state}, state)
```
Comment (mat-hek) [inline lib/membrane/core/element.ex]: you could just do `noreply(result, state)`

**Review by mat-hek — CHANGES_REQUESTED** [review]

**Review by mat-hek — APPROVED** [review]

**Pull request merged**